packages/server/src/project/dto/project.dto.ts

```diff
@@ -83,3 +100,27 @@ export class ConfigurableProjectSettings implements Partial<Omit<Project, 'id' |
   @Field({ nullable: true })
   redirectUrl: string | null;
 }
+
+/**
+ * Input type for updating a project's settings
+ */
+@InputType()
+export class ProjectSettingsInput implements Partial<ProjectSettingsModel> {
+  @IsOptional()
+  @IsNotEmpty()
+  @Field()
```

Comment:
For optional fields, `@Field` should be updated to look something like `@Field({ nullable: true })` and for the type something like `displayProjectName?: boolean` so typescript can enforce that `displayProjectName` may not be defined.